R/addWaterYear.R

```diff
@@ -36,35 +38,45 @@ addWaterYear <- function(rawData){
   # set the name of the date column(s) to use for calculating the WY
   dateColName <- names(rawData)[names(rawData) %in% allowedDateColNames]
   
-  # set up new df that will include water year columns
-  wyData <- rawData
-  
   for(dateCol in dateColName){
-    dateColWY <- paste0(dateCol, "WY")
+    dateColWY <- allowedWYColNames[dateCol]
     
     # if this WY column already exists, do not add another
-    if(dateColWY %in% names(wyData)){
+    if(dateColWY %in% names(rawData)){
```

Comment:
I think you could get this logic in line 39 too (?):

Something like (untested!!!!):
```
dateColName <- names(rawData)[names(rawData) %in% allowedDateColNames]
dateColNames <- dateColNames[!(dateColNames %in% names(rawData))]
```
or put an `&` directly in 39. 

In this case, we can totally keep that `if/next`, since at most, this loop is happening 4 times. In other loops that could be happening for 100's or 1000's or more times, I'm pretty sure a `if/next` statement is pretty slow compared to doing it ahead of time outside of the loop. That's why I always try to see if I can get them out just as best practice.